### PR TITLE
Enhance test system and add migration tests

### PR DESCRIPTION
**Pull request by philpep**

### CODE REVIEW

File: run-tests.py [hunk 1]
```diff
@@ -77,9 +77,12 @@ def test(ctx, image):
     tag = get_tag(image, True)
     if not image_exists(tag):
         ctx.invoke(build, image=image, salt=True)
+    postgres_tag = get_tag("postgres", False)
+    if not image_exists(tag):
+        ctx.invoke(build, image="postgres", salt=False)
 
     import pytest
-    return pytest.main(["--docker-image", tag] + ctx.args)
+    return pytest.main(["--docker-image", tag, "--postgres-image", postgres_tag] + ctx.args)
```
Comment (dlax) [inline run-tests.py]: if `postgres_tag is False`, the option should not be passed, no?

File: test/test_migration.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        pkg = "hg+http://hg.logilab.org/master/cubes/saem_ref@default#egg=cubicweb-saem-ref"
+    else:
+        pkg = "hg+http://hg.logilab.org/review/cubes/saem_ref@{0}#egg=cubicweb-saem-ref".format(
+            saem_ref_upgrade_revision)
+    run("pip install %s", pkg)
+    run("su - saemref -c 'CW_MODE=user cubicweb-ctl db-restore saemref /dumps/{0}'".format(dump))
+    out = Command("su - saemref -c 'CW_MODE=user cubicweb-ctl upgrade -v 0 --force --backup-db=n saemref'")
+    # upgrade exit 0 even if a migration failed
+    assert out.rc == 0
+    assert "-> instance migrated." in out.stdout
```
Comment (dlax) [inline test/test_migration.py]: add `, out.stdout` (possibly decoded to get a readable multiline string) to last assert

**Comment by dlax** [comment]
would have nice to have this handled through a Salt state, but that's a first step.

File: README.rst [hunk 3]
```diff
@@ -84,7 +84,19 @@ The command ``./run-test.py test centos6`` built a provisioned image
...
 
 Note that you can add any parameters that testinfra (pytest) accept, for
-instance ``./run-test.py test centos6 -k idempotence --pdb``.
+instance ``./run-test.py test centos6 -- -k idempotence --pdb``.
+
+
+Testing migration
+=================
+
+Add dumps generated by ``cubicweb-ctl db-restore`` in the ``test/dumps`` directory, then::
```
Comment (philpep) [inline README.rst]: s/db-restore/db-dump

File: run-tests.py (hunk 1 above)
Comment (philpep) [inline run-tests.py]: postgres_tag come from `get_tag("postgres", False)` and is always `"saemref-formula:postgres"``

**Pull request merged**